### PR TITLE
feat: add flag to CacheControlInstrumentation to optionally allow max-age of zero

### PR DESCRIPTION
**Pull request by trophied**
This change adds an optional flag to `CacheControlInstrumentation`, `allowZeroMaxAge`, to allow consumers to configure the behavior such that resolved header can validly resolve a header value of `max-age=0`.

### CODE REVIEW

**Review by dariuszkuc — APPROVED** [review]

**Comment by dariuszkuc** [comment]
Linter failed -> can you run `./gradlew spotlessApply` (this should apply auto format, there might be other linter issues that need manual fixing).

**Comment by trophied** [comment]
> Linter failed -> can you run `./gradlew spotlessApply` (this should apply auto format, there might be other linter issues that need manual fixing).

Sure thing, I must have missed re-running after changing some stuff out 👍

File: graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/caching/CacheControlInstrumentation.java [hunk 1]
```diff
@@ -223,11 +238,13 @@ void replace(@Nullable CacheControlScope scope) {
 
     public Optional<String> maybeAsString() {
       Integer maxAgeValue = maxAge == null ? 0 : maxAge;
-      if (maxAgeValue.equals(0)) {
+
+      if (!allowZeroMaxAge && maxAgeValue.equals(0)) {
```
Comment (dariuszkuc) [inline graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/caching/CacheControlInstrumentation.java:242]: actually I think we should change the logic here -> we should detect between `maxAge` is `null` and `0` so I think we probably should have the condition as

```java
if (maxAge == null || (!allowZeroMaxAge && maxAge == 0)) {
  return Optional.empty();
}
```

**Review by dariuszkuc — CHANGES_REQUESTED** [review]

File: graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/caching/CacheControlInstrumentation.java (hunk 1 above)
Comment (trophied) [reply graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/caching/CacheControlInstrumentation.java:242]: Sounds good, I have added the explicit handling of `null` 👍

**Review by dariuszkuc — APPROVED** [review]

**Pull request merged**